pkg/drivers/vfkit/driver_darwin.go

```diff
@@ -466,10 +469,35 @@ func (d *Driver) sendSignal(s syscall.Signal) error {
 	return proc.SendSignal(s)
 }
 
+var (
+	once              sync.Once
+	macCurrentVersion string
+	errGettingVersion error
+)
+
 func (d *Driver) supportsVirtiofs() bool {
-	macosVersion, _, err := crcos.RunWithDefaultLocale("sw_vers", "-productVersion")
-	if err != nil {
+	once.Do(func() {
+		macCurrentVersion, errGettingVersion = syscall.Sysctl("kern.osproductversion")
+		logging.Debugf("kern.osproductversion is: %s", macCurrentVersion)
+	})
+	if errGettingVersion != nil {
 		return false
 	}
-	return strings.HasPrefix(macosVersion, "12.")
+	supportsVirtioFS, err := macosAtLeast(macCurrentVersion, "12.0.0")
+	if err != nil {
+		log.Debugf("Not able to compare version: %v", err)
+	}
+	return supportsVirtioFS
+}
+
+func macosAtLeast(curVersion string, targetVersion string) (bool, error) {
+	cVersion, err := semver.NewVersion(curVersion)
+	if err != nil {
+		return false, errors.Wrap(err, fmt.Sprintf("cannot parse %s", curVersion))
+	}
+	tarVersion, err := semver.NewVersion(targetVersion)
```

Comment:
`tarVersion` makes me think of the `tar` archiving format, not of target. You could go with `targetVersionStr` and `targetVersion`.